scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/Synchronization.java

```diff
@@ -0,0 +1,1135 @@
+     *
+     * For example, to append a {@code String} msg to a value if the value is already defined, and otherwise set msg as the new mapping
+     * mapping:
+     *
+     * <pre> {@code
+     * store.conditionalCompute(channel, key, "{ k, v -> v != null}", "{k, v -> v.concat(\"" + msg + "\")}", "{k, v -> \"" + msg + "\""}")}</pre>
+     *
+     * <p>If either function returns {@code null}, the mapping is removed (or
+     * remains absent if initially absent).  If the predicate or function itself throw an
+     * (unchecked) exception, the exception is rethrown as a ClosureEvaluationException, and the current mapping
+     * is left unchanged.
+     *
+     * The default implementation is equivalent to performing the following
+     * steps for this {@code map}, then returning a PredicateActionResult object containing the current value or
+     * {@code null} if absent:
+     *
+     * <pre> {@code
+     * V oldValue = map.get(key);
+     * if (predicate.test(key, oldValue)) {
+     *  map.compute(key, thenRemappingFunction);
+     * } else {
+     *  map.compute(key, elseRemappingFunction);
+     * }
+     * }</pre>
+     *
+     * <p>This implementation guarantees that the operation is performed atomically on the channel map.
+     *
+     * @param channel channel identifier
+     * @param key key with which the specified value is to be associated
+     * @param thenRemappingFunction the function to compute a value if the predicate is true
+     * @param elseRemappingFunction the function to compute a value if the predicate is false
+     * @return a {@link PredicateActionResult} object which contains the result of the predicate evaluation and either (true) the new value associated with the key after applying thenRemappingFunction, or (false) the new value associated with the key after applying elseRemappingFunction.
+     * @throws NullPointerException if the specified key is null and
+     *         this map does not support null keys, or the
+     *         remappingFunction is null
+     * @throws CompilationException if an error occurred during the compilation of the specified action
+     * @throws ClosureEvaluationException if an error occurred during the evaluation of the specified action
+     * @throws InvalidChannelException if the channel does not exist
+     * @throws IOException if an error occurred when persisting the modified channel on disk
+     * @since 1.8
+     * @see java.util.Map#compute(Object, BiFunction)
+     * @see java.util.function.BiPredicate
+     * @see java.util.function.BiFunction
+     * @see <a href="http://groovy-lang.org/closures.html">Groovy Closures</a>
+     * @see org.ow2.proactive.scheduler.synchronization.PredicateActionResult
+     */
+    PredicateActionResult conditionalCompute(String channel, String key, String predicate, String thenRemappingFunction,
+            String elseRemappingFunction) throws InvalidChannelException, CompilationException, IOException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)